testing/jormungandr-integration-tests/src/jormungandr/transactions.rs

```diff
@@ -89,3 +93,124 @@ pub fn accounts_funds_are_updated_after_transaction() {
         "receiver value after transaction"
     );
 }
+
+#[test]
+fn expired_transactions_rejected() {
+    let receiver = startup::create_new_account_address();
+    let mut sender = startup::create_new_account_address();
+
+    let (jormungandr, _) = startup::start_stake_pool(
+        &[sender.clone()],
+        &[receiver.clone()],
+        ConfigurationBuilder::new()
+            .with_slots_per_epoch(30)
+            .with_consensus_genesis_praos_active_slot_coeff(ActiveSlotCoefficient::MAXIMUM)
+            .with_slot_duration(1)
+            .with_linear_fees(LinearFee::new(0, 0, 0))
+            .with_mempool(Mempool {
+                pool_max_entries: 1_000.into(),
+                log_max_entries: 1_000.into(),
+                persistent_log: None,
+            }),
+    )
+    .unwrap();
+
+    let jcli = JCli::default();
+
+    let valid_transaction = sender
+        .transaction_to(
+            &jormungandr.genesis_block_hash(),
+            &jormungandr.fees(),
+            chain_impl_mockchain::block::BlockDate::first(),
```

Comment:
mmm, I think there's still some date magic under this and I don't particularly like it. Didn't we fix it already (or was it something different)? IIRC we agreed to not have date manipulation, at least for interfaces which are simply building a transaction like you would on the command line like `Wallet`

Reply:
Yes that is correct. In this case the magic is hidden in `jormungandr_testing_utils::testing::fragments::FragmentBuilder`. We have #3529 which should clean these up, so I didn't bother fixing them again. We can rebase this branch onto it or the other way around.

Reply:
Indeed, I recalled seeing a PR for that. All good then! But I would wait for it to land before merging this PR, as it will require changes to the tests I guess